genie-web/src/main/java/com/netflix/genie/web/agent/resources/AgentFileProtocolResolver.java

```diff
@@ -107,12 +121,44 @@ public Resource resolve(final String location, final ResourceLoader resourceLoad
             return null;
         }
 
+        final Pair<Integer, Integer> range = parseRangeHeader(uri.getFragment());
+        final Integer rangeStart = range.getLeft();
+        final Integer rangeEnd = range.getRight();
+
         final AgentFileStreamService.AgentFileResource resourceOrNull =
-            agentFileStreamService.getResource(jobId, relativePath, uri).orElse(null);
+            agentFileStreamService.getResource(jobId, relativePath, uri, rangeStart, rangeEnd).orElse(null);
 
         if (resourceOrNull != null) {
             log.debug("Returning resource: {}", resourceOrNull.getDescription());
         }
         return resourceOrNull;
     }
+
+    private Pair<Integer, Integer> parseRangeHeader(final String rangeHeader) {
```

Comment:
I thought (in looking at some of this in the past) that Spring had a bunch of utilities for working with headers and handling range requests. I know there is a [HttpRange](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/HttpRange.html) abstraction that exists already. Not sure if it would help to get rid of all this error checking and whatnot but thought I'd point it out to save you from possibly reinventing the wheel.

Reply:
Yes! This does pretty much everything in the PR i just wrote. I got busy rewriting the missing commit and forgot to do some research to see if something was available off the shelf..
I'll replace with this, should be pretty quick to do so.